packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -366,24 +374,6 @@ const nodeProxyHandler: AdaptingProxyHandler<NodeProxyTarget, EditableTree> = {
 			const fieldKey: FieldKey = brand(key);
 			const fieldSchema = target.getFieldSchema(fieldKey);
 			const multiplicity = target.lookupFieldKind(fieldKey).multiplicity;
-			if (target.has(fieldKey) && isPrimitiveValue(value)) {
```

Comment:
this code updated the value of the existing node instead of replacing it (which is what the below code does). Having this extra codepath here makes the behavior more complex and is not needed.

This is a behavior change since the new code object to assigning a field to a primitive value when there are multiple types of primitive nodes that could be used with that value. The old code would handle this case by reusing the existing node, and assigning its value. This slightly changes merge semantics in an edge case and makes assigning over a node depend on the type of node that its replacing not just the field its assigning into.